src/go/rpk/pkg/api/admin/api_cloud_storage_test.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package admin
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type (
+	testRequestHandler func(w http.ResponseWriter, r *http.Request)
+	testFn             func(t *testing.T) testRequestHandler
+	requestTestCase    struct {
+		name             string
+		testFn           testFn
+		expectedResponse interface{}
+	}
+)
+
+func TestStartAutomatedRecovery(t *testing.T) {
+	successfullStartResponse := RecoveryStartResponse{
+		Code:    200,
+		Message: "Automated recovery started",
+	}
+
+	runTest := func(t *testing.T, test requestTestCase) {
+		baseURL := "http://non-existent-url"
+
+		if test.testFn != nil {
```

Comment:
testFn must be non-nil as implemented, this conditional can be removed and baseURL can be dropped in favor of using server.URL directly.

If testFn were non-nil, this test would fail on line 52 anyway because the client would have nowhere to issue the request.

Reply:
I was apparently a bit to quick with the changes I made in https://github.com/redpanda-data/redpanda/commit/ae89dd9cd57011fcd340fc1a2fbd339bdec18733 and forgot to go through how it impacted other parts of the code. Good catch!